README.md
# Terminal Designer

    This is a Terminal designer like(Text Processing) ForeGround, BackGround we can change the colors in 3 different ways using this module.

## Install the Package from python
    https://pypi.org/project/TerminalDesigner/0.0.1/
    
(or)

    pip install TerminalDesigner==0.0.1

After installation you can import the package using

    import Designer

## There are 7 python files.
- BackGroundColor.py
- Color.py
- ForeGroundColor.py
- General.py
- HexName.py
- RandomColor.py
- Values.py

There are **255** different shades with **610** colors combinations. these colors are handle with **6** Modules :-
- BackGroundColor.py
- Color.py
- ForeGroundColor.py
- General.py
- HexName.py
- RandomColor.py
_____________________________________________________________________________

# 1.BackGroundColor Module
it have **610** color named functions and set_default_stylebg() extra one of the function.
>Note : this function names are have colors name + bg
**for example : redbg("hello"),bluebg("world")**

It is easy to use we just import the BackGroundColor Function From Designer and call the names of the colors.
It will return the colored BackGround text.
## How to use functions in the module:
* **text** = it's need a string input like ['text' (or) "text"]
* **Style** :-
    - Normal     = 0
    - Bold       = 1
    - Light      = 2
    - Italicized = 3
    - UnderLined = 4        
    - Blink      = 5
* ## Example :
    >**Nameof theFunction**("Hello World", Style = 0 ) it's return color text without any style    

    >**Nameof theFunction**("Hello World", Style = 1 ) it's return color text with bold text       

    >**Nameof theFunction**("Hello World", Style = 2 ) it's return color text with light text      

    >**Nameof theFunction**("Hello World", Style = 3 ) it's return color text with Italicized style

    >**Nameof theFunction**("Hello World", Style = 4 ) it's return color text with UnderLine       

    >**Nameof theFunction**<("Hello World", Style = 5 ) it's return Blinking  color text             


## for example:
        #import the all functions BackGroundColor Module then just call

        print(redbg("hello")+bluebg("world"))
        print(brownbg("it's"),pinkbg("realy"),greenbg("interesting"))
        print(redbg("hello"+pinkbg("world")))
## BackGround Module set_default_stylebg() function
  >its,s need only one parameter String type it's return type is None.

  >it's used to change the default values (or) change the value of style to all colors
 ## Ex : set_default_stylebg(3) then it chenge the Italicized style to all color_function
----------------------------------------------------
* Some style propertys
    - Normal     = 0
    - Bold       = 1
    - Light      = 2
    - Italicized = 3
    - UnderLined = 4        
    - Blink      = 5

> **Return**: it's return None Value,it's not return any values just it change the changes in BackGroundColor file.
        
---------------------------------------------------------------------------

# 2.ForeGroundColor Module
    

it have **610** color named functions and set_default_style() extra one of the function.
It is easy to use we just import the ForeGroundColor Function From Designer and call the names of the colors.
It will return the colored ForeGround text.
## How to use functions in the module:
* **text** = it's need a string input like ['text' (or) "text"]
* **Style** :-
    - Normal     = 0
    - Bold       = 1
    - Light      = 2
    - Italicized = 3
    - UnderLined = 4        
    - Blink      = 5
* ## Example :
    >**Nameof theFunction**("Hello World", Style = 0 ) it's return color text without any style    

    >**Nameof theFunction**("Hello World", Style = 1 ) it's return color text with bold text       

    >**Nameof theFunction**("Hello World", Style = 2 ) it's return color text with light text      

    >**Nameof theFunction**("Hello World", Style = 3 ) it's return color text with Italicized style

    >**Nameof theFunction**("Hello World", Style = 4 ) it's return color text with UnderLine       

    >**Nameof theFunction**<("Hello World", Style = 5 ) it's return Blinking  color text             


## for example:
        #import the all functions BackGroundColor Module then just call

        print(red("hello")+blue("world"))
        print(brown("it's"),pink("pink"),green("interesting"))
        print(red("hello"+pink("world")))
# BackGround Module set_default_style() function
  >its,s need only one parameter String type it's return type is None.

  >it's used to change the default values (or) change the value of style to all colors
 ## Ex : set_default_stylebg(3) then it change the Italicized style to all color_function
----------------------------------------------------
* Some style propertys
    - Normal     = 0
    - Bold       = 1
    - Light      = 2
    - Italicized = 3
    - UnderLined = 4        
    - Blink      = 5

> **Return**: it's **return None** Value,it's not return any values just it change the changes in BackGroundColor file.

# 3. HaxName Module

> **Note**: it's just like ForeGround Module these function names are hax value names.

it have **610** color named functions and *set_default_style()* extra one of the function.
It is easy to use we just import the ForeGroundColor Function From Designer and call the names of the colors.
It will return the colored ForeGround text.
## How to use functions in the module:
* **text** = it's need a string input like ['text' (or) "text"]
* **Style** :-
    - Normal     = 0
    - Bold       = 1
    - Light      = 2
    - Italicized = 3
    - UnderLined = 4        
    - Blink      = 5
* ## Example :
    >**Nameof theFunction**("Hello World", Style = 0 ) it's return color text without any style    

    >**Nameof theFunction**("Hello World", Style = 1 ) it's return color text with bold text       

    >**Nameof theFunction**("Hello World", Style = 2 ) it's return color text with light text      

    >**Nameof theFunction**("Hello World", Style = 3 ) it's return color text with Italicized style

    >**Nameof theFunction**("Hello World", Style = 4 ) it's return color text with UnderLine       

    >**Nameof theFunction**<("Hello World", Style = 5 ) it's return Blinking  color text             


## for example:
        #import the all functions BackGroundColor Module then just call

        print(red("hello")+blue("world"))
        print(brown("it's"),pink("relay"),green("interesting"))
        print(red("hello"+pink("world")))
# BackGround Module set_default_style() function
  >its,s need only one parameter String type it's return type is None.

  >it's used to change the default values (or) change the value of style to all colors
 ## Ex : set_default_stylebg(3) then it change the Italicized style to all color_function
----------------------------------------------------
* Some style propertys
    - **Normal**     = 0
    - **Bold**       = 1
    - **Light**      = 2
    - **Italicized** = 3
    - **UnderLined** = 4        
    - **Blink**      = 5

> **Return**: it's **return None** Value,it's not return any values just it change the changes in BackGroundColor file.
-----------------------------------------------------

# 4.RandomColor Module
There are 6 ( *functions frd2b, frb2d, brd2b, brb2d, bgrandomclr, fgrandomclr* ) this function are performance are based on random.

* ## fgrandomclr()
    it's return the random colored ForeGround text(stringType) based on given text.
    >def fgrandomclr( String : str ) -> str :

    <h2 style="color:red" >import  the RandomColor Module from Designer

        Example:
        fgrandomclr("The")
* ## bgrandomclr()
    it's return the random colored BackGround(stringType) text based on given text.
    >def bgrandomclr( String : str ) -> str :
        
        Example:
             bgrandomclr( "world" ):
* ## brb2d()
    its's return the bright to dull color given text for BackGround.
    >def brb2d(String : str) -> str:

        Example:
             print(brb2d("is"))
* ## brd2b()
    its's return the dull to bright color given text for BackGround.

    >def brd2b(String : str) -> str:

        Example:
            print(brd2b("Fake"))
* ## frb2d()
    its's return the bright to dull color given text for ForeGround
    >def frb2d(String : str) -> str:

        Example:
            print(frb2d("hello"))
* ## frd2b()
    its's return the dull to bright color given text for ForeGround
    >def frd2b(String : str) -> str:

        Example:
            print(frd2b("hello"))

-----------------------------------------------------

# 5.General Module
This Model have generally used functions. it's have**one class** and **14** functions:
* # class
    * # image_
        it's need to file path of the image.
            
            Example :
            image_ = image_("/file/path/file.jpg")
       *  ## rgb_value()
            it's return the Pixels of the image.
            > def rgb_value(self) -> list :
                Example :
                    print(image_.rgb_value())
        * ## img_size()
            it's return the size of the image.
            >def img_size(self) -> tuple:
                Example:
                    print(image_.img_size())
        * ## used_colors() 
            it's return the tuple.the tuple have used colors in the image.
            >def used_colors(self,graph_view= False, return_graph_view = False, graph_view_with_rgb=False,view_text="") -> Union[str,tuple]:
            
            ## Parameters
            * **graph_view** - if the graph_view is true it print the colors in the terminal
            * **return_graph_view** - if the return_graph_view is true it return the colors value.
            * **graph_view_with_rgb** - if graph_view_with_rgb is true print or return the rgb values with bg color
            * **view_text** - view_text is used to print the given text with used background.
                    Example:
                        image_.used_colors()
                        #it return the used colors in tuple type.
        * # most_used_color()
            it return the tuple it have most used color which is rgb.
            >def most_used_color(self) -> str:
                Example:
                    image_.most_used_color()
                    # it return most  used color of rgb value which is tuple.

* # Functions
    * # img_to_ascii()
        it's very interesting module,it's convert image into ascii code.the return type is string.
        >def img_to_ascii(Path) -> str :

            Example :
                path = "/file/patch/image.jpg " 
                img_to_ascii()
                # it's return string which is replicate image.
    * # start()
         it's apply the given colors in terminal which is after printing this start function.it's until apply the given color to text until will call the end function.
        >def start( String : str ,Type_ : str ="fg", style : int = 0) -> str :
         ## Parameters
        * **String** = it's need a string type input.
        * **Type_** : it's need the only two input that is nether "bg" nor "fg".
        * **style** : it's need an int type input.
            ## values 
            - Normal     = 0
            - Bold       = 1
            - Light      = 2
            - Italicized = 3
            - UnderLined = 4        
            - Blink      = 5
                    Example:
                        #import all the functions from General model
                        start("red")
                        #its apply the red color after printing text in the terminal
    * # stop()
        it's stop the applying colors which is performed by start.
        >def stop() -> str :

            Example:
                start("blue")
                stop()
                #it's apply the blue color to the text between start() and end().
    * # clrscreen()
        it just clear the terminal's screen.
        >def clrscreen() -> None :
        
            Example:
                #import module & call
                clrscreen()
                #after calling screen will clear.
    
    * # Fullbgclr()
        it's apply background and foreground color to total terminal screen based on given parameters.
        >def Fullbgclr( FColor : str, BColor :str ) -> None:
        ## Parameters
        * FColor = it's need the string value to__ForeGround__ of the text.
        - BColor = it's need the string value to__BackGround__ of the text.
        ## These values are suitable inputs :
        * `{'BLACK': 0, 'BLUE': 1, 'GREEN': 2,'AQUA': 3, 'RED': 4, 'PURPLE': 5, 'YELLOW':6, 'WHITE ': 7, 'GRAY': 8, 'LIGHT BLUE': 9,'LIGHT GREEN': 'A', 'BRIGHT WHITE': 'F','LIGHT AQUA': 'B', 'LIGHT YELLOW': 'E','LIGHT RED': 'C', 'LIGHT PURPLE': 'D'}`
        * `1, 2, 3, 4, 5, 6, 7, 8, 9, 'A', 'B','C', 'D',' E', 'F'`

                Example:
                    Fullbgclr("red","black")
    * # color()
        it apply the color given string and just return the rgb value of the given text.
        >def color(TColor : str, String:str)->str:

        # Parameters
        Tolor : it need a Color name to apply.
        String : it's need a string value

            Example : 
                color("red","hello")
    
     * # image_art()
        it's also very interesting module,it's convert image into colored text code.the colored text are replicate the image in terminal.the return type is string.
        >Note : it's Only get JPG File and using below ( 150 X 150 )px image is more suitable
        function :
        >def img_to_ascii(Path) -> str :
        ## Parameters
        - Character = it takes string (or)   - character using this output are replicate given image
        - Path = it's need Path of the file
        - Type = it's need inputs like ForeGround["fg"] (or) BackGround["bg"] so based on that the output replicate

                Example :
                    image_art(Character : str, Path: str)
                    image_art( "*", /../../)
                    path = "/file/patch/image.jpg " 
                    # it's return colored string which is replicate image.
    * ## rgb_to_txt
        it's used to print the colored text based on given rgb value
        >Note : it's take single (r,g,b) value.
        >def rgb_to_txt(text : str, rgb : tuple style : int = 0,type_ : str ="fg") ->  str:
        
            Example:
                rgb_to_txt("str",(255,255,255))
    * ## rgb_to_name()
        there is the module under development so the rgb values in data it's return name of the data else it return 0.
        
        >def rgb_to_name(rgb_val : tuple) -> Union[str,int] :
        
            Example:
                rgb_to_name((0,0,0))
        
    * # name_to_rgb()
        there is the module under development so the given name in data it's return value of the data else it return None. 
        >def name_to_rgb(Name_of the_color : str )-> tuple :

            Example:
                name_to_rgb("red")
    
    * # hex_to_name()
        there is the module under development so the given hex in data it's return value of thr data else it return None.
        >def hex_to_name(Name_of the_color : str )-> str:   
            Example:

                hex_to_name("#FFFFFF")
    * # name_to_hex()
        there is the module under development so the given name in data it's return value of the data else it return None.
        > name_to_hex(hex_val : str ) -> str:

            Example:
            name_to_hex("green")
    * # rgb_to_hex()
        there is the module under development so the given rgb in data it's return value of the data else it return None. 
        >def rgb_to_hex(rgb_val : tuple ) -> str:
        
            Example:
                rgb_to_hex((0,0,0))
    * # hex_to_rgb()
        there is the module under development so the given hex in data it's return value of the data else it return None. 
        >def hex_to_rgb(rgb_val : str ) -> tuple:

            Example:
                hex_to_rgb("#FFFFFF")

-------------------------------------------------------

# 6.Color Module
 it have one class that is **Colors**.
 Color's have two functions. 
 * fore_ground_color
 * back_ground_color 
     -----------------------------------------
     * ## fore_ground_color()
         it's return the color code of the given text. it's return the color code based on given RGB.
         >fore_ground_color(text : str, r : int, g : int, b : int, style : int = 0)
         # Parameters
          * text = it's need a string input\n
          *  r, b, g = This is the RGB color code like white RGB(255,255,255)\n
          * # Style:
             - Normal     = 0
             - Bold       = 1
             - Light      = 2
             - Italicized = 3
             - UnderLined = 4        
             - Blink      = 5
         
             Example :
                 #import the function from colors module
                 fore_ground_color("hello",0,0,0,2)
     --------------------------------------------
     * ## back_ground_color()
         it's return the BackGround color code of the given text. it's return the color code based on given RGB.
         > back_ground_color(text : str, r : int, g : int, b : int, style : int = 0)
         # Parameters
          * text = it's need a string input\n
          *  r, b, g = This is the RGB color code like white RGB(255,255,255)\n
          * # Style:
             - Normal     = 0
             - Bold       = 1
             - Light      = 2
             - Italicized = 3
             - UnderLined = 4        
             - Blink      = 5
         
             Example :
                 #import the function from colors module
                 back_ground_color("black",0,0,0,3)
------------------------------------------------------
* # 7.Values
    this is have the colors and rgb values to the variables.
    ## variables
    - ValuesNRH
    - hex_name
    - name_hex 
    - tuple_name
    - rgb
    - colors
----------------------------------------
# Performance
 it perform terminal color operations.
 ## Like
 - ForeGround
 - BackGround
 
 Each module are perform each works.

-----------------------------------------
# Author
**Name** : Nagi Pragalatha.N

**Email** : nagipragalathan@gmail.com

**About** : https://nagipragalathan.github.io/Personal_website/about.html

**GitHub** :  https://github.com/NagiPragalathan/

**Contact** : [InstaGram](https://instagram.com/mr_lazy_05?igshid=YmMyMTA2M2Y=),[FaceBook](https://www.facebook.com/profile.php?id=100081414617956),[LinkedIn](https://www.linkedin.com/in/nagi-pragalathan-n-a03a55230)

--------------------------------------------
# Package
**Name** : Designer

**Version** : 0.0.0.1

**License** : MIT

--------------------------------------------

# Thank you for visiting--

--------------------------------------------


#  BackGroundColor Module's functions


CHANGELOG.txt
Change Log
==========

0.0.0.1 (04/08/2022)
--------------------

-First Release

Designer/General.py
from .ForeGroundColor import *
from .Color import colors
from PIL import Image
from .Values import tuple_name, hex_name, name_hex
import os


ASCII_CHARS = ["@", "#", "S", "%", "?", "*", "+", ";", ":", ",", ".","A","B","C","D","E","F","G","H","I"]
ListOfColors : list = list(rgb.keys())
Type = ""


class image_:
    path = ""
    def __init__(self,Path : str) -> None :
        self.path = Path
    def rgb_value(self) -> list :
        """ it's return the Pixles of the image """
        self.img = Image.open(self.path)
        pix = list(self.img.getdata())
        return pix
    def img_size(self) -> tuple:
        """it's return the size of the image return type tuple"""
        self.img = Image.open(self.path)
        width, height = self.img.size
        return (width, height)
    
    def used_colors(self,graph_view=False,return_graph_view=False,graph_view_with_rgb=False,view_text=".") -> Union[str,tuple]:
        """ it's return the tuple.the tuple have used colors in the image 
            if the graph_view is true it print the colors in the terminal.\n
            if the return_graph_view is true it return the colors value.\n
            if graph_view_with_rgb is true print or return the rgb values with bg color.\n
            view_text is used to print the given text with used background. 
            """
        
        rgb_val : list = self.rgb_value()
        rm_dup = list(set(rgb_val))
        width = os.get_terminal_size().columns
        widthCenterPoint = int(width/2)
        if(graph_view):
            if return_graph_view:
                str_ = ""
                for x,i in enumerate(rm_dup):
                    if graph_view_with_rgb:
                        str_= str_+rgb_to_txt(f"{i}",i,type_="bg")
                    elif(graph_view_with_rgb==None):
                        str_= str_+rgb_to_txt(f"{x}",i,type_="bg")
                    else:
                        str_= str_+rgb_to_txt(f"{view_text}",i,type_="bg")
                return str_
            else:
                for x,i in enumerate(rm_dup):
                    if graph_view_with_rgb:
                        print(rgb_to_txt(f"{i}",i,type_="bg"),end="")
                    elif(graph_view_with_rgb==None):
                        print(rgb_to_txt(f"{x}",i,type_="bg"),end="")
                    else:
                        print(rgb_to_txt(f"{view_text}",i,type_="bg"),end="")
        else:
            return rm_dup
    def most_used_color(self) -> str:
        """it return the tuple it have most used color which is rgb."""
        rgb_val : list = self.rgb_value()
        rm_dup = list(set(rgb_val))
        for i in rm_dup:
            max=0
            return_ = tuple()
            if(max < rgb_val.count(i)):
                max = rgb_val.count(i)
                return_ = i
        return return_

def img_to_ascii(Path) -> str :
    """ it's convert image to ascii code """
    a=Image.open(Path).convert('L')
    pixels = a.getdata()
    characters = "".join([ASCII_CHARS[pixel//25] for pixel in pixels])
    return characters

def start( String : str ,Type_ : str = "fg", style : int = 0) -> str :
    """ it's apply the given colors in terminal which is after printing this start function.it's until apply the given color to text until will call the end function.
        >def start( String : str ,Type_ : str ="fg", style : int = 0) -> str :
         ## Parameters
        * **String** = it's need a string type input.
        * **Type_** : it's need the only two input that is nether "bg" nor "fg".
        * **style** : it's need an int type input.
            ## values 
            - Normal     = 0
            - Bold       = 1
            - Light      = 2
            - Italicized = 3
            - UnderLined = 4        
            - Blink      = 5
                    Example:\n
                        #import all the functions from General model\n
                        start("red") \n
                        #its apply the red color after printing text in the terminal
    """
    global Type
    Type = Type_
    value : list = colors.get(String.upper())
    if( Type == "fg" ):
        print("\033[{};38;2;{};{};{}m' '".format( style, value[0], value[1], value[2]))
        return "\033[{};38;2;{};{};{}m' '".format( style, value[0], value[1], value[2])
    else:
        print("\033[{};48;2;{};{};{}m' '".format( style, value[0], value[1], value[2]))
        return "\033[{};48;2;{};{};{}m' '".format( style, value[0], value[1], value[2])
    
def stop() -> str :
    """it's stop the applying colors which is performed by start.\n
        Example:\n
                    start("blue")\n
                    stop()\n
                    #it's apply the blue color to the text between start() and end().
    """
    if( Type == "fg" ):
        print("\033[0;38;0;255;255;255m")
        return "\033[0;38;0;255;255;255m"
    else:
        print("\033[0;48;0;255;255;255m")
        return "\033[0;48;0;255;255;255m"
    
def clrscreen() -> None :
    """ ## it just clear the terminal's screen.\n
        >def clrscreen() -> None :
        
            Example:
                #import module & call
                clrscreen()
                #after calling screen will clear."""
    print("clearing...")
    os.system("clear || cls")

def Fullbgclr( FColor : str, BColor : str ) -> None:
    """it's apply background and foreground color to total terminal screen based on given parameters.\n
        >def Fullbgclr( FColor : str, BColor :str ) -> None:\n
        ## Parameters
        * FColor = it's need the string value to__ForeGround__ of the text.
        - BColor = it's need the string value to__BackGround__ of the text.
        ## These values are suitable inputs :
        * `{'BLACK': 0, 'BLUE': 1, 'GREEN': 2,'AQUA': 3, 'RED': 4, 'PURPLE': 5, 'YELLOW':6, 'WHITE ': 7, 'GRAY': 8, 'LIGHT BLUE': 9,'LIGHT GREEN': 'A', 'BRIGHT WHITE': 'F','LIGHT AQUA': 'B', 'LIGHT YELLOW': 'E','LIGHT RED': 'C', 'LIGHT PURPLE': 'D'}`
        * `1, 2, 3, 4, 5, 6, 7, 8, 9, 'A', 'B','C', 'D',' E', 'F'`

                Example:
                    Fullbgclr("red","black")"""
    Values : dict = {'BLACK': 0, 'BLUE': 1, 'GREEN': 2, 'AQUA': 3, 'RED': 4, 'PURPLE': 5, 'YELLOW': 6, 'WHITE ': 7, 'GRAY': 8, 'LIGHT BLUE': 9, 'LIGHT GREEN': 'A', 'BRIGHT WHITE': 'F', 'LIGHT AQUA': 'B', 'LIGHT YELLOW': 'E', 'LIGHT RED': 'C', 'LIGHT PURPLE': 'D'
              ,"0":"0","1":"1","2":"2","3":"3","4":"4","5":"5","6":"6","7":"7","8":"8","9":"9","A":"A","F":"F","B":"B","E":"E","C":"C","D":"D"}    
    try:
        fval : str = Values.get(str(FColor).upper())
        bval : str = Values.get(str(BColor).upper())
        os.system("color "+str(fval)+str(bval))
    except:
        raise ValueError("""Undefind value (non-listed value) select values between { 0 : "Black", 1 : "Blue", 2 : "Green", 3 : "Aqua", 4 : "Red", 5 : "Purple", 6 : "Yellow", 7 : "White ", 8 : "Gray", 9 : "Light Blue", "A" : "Light Green", "F" : "Bright White", "B" : "Light Aqua", "E" : "Light Yellow", "C" : "Light Red", "D" : "Light Purple"}""")

def color(TColor : str, String:str)->str:
    """it apply the color given string and just return the rgb value of the given text.\n
        >def color(TColor : str, String:str)->str:

        # Parameters
        
        Tolor : it need a Color name to apply.
        String : it's need a string value

            Example : 
                color("red","hello")"""
    value : list = list(colors.get(TColor.upper()))
    return "\033[0;38;2;{};{};{}m{}\033[0;38;0;255;255;255m".format(value[0], value[1], value[2], String)

def rgb_to_txt(text : str, rgb : tuple,style : int = 0,type_ : str ="fg") ->  str:
    """ it's used to print the colored text based on given rgb value\n
        Note : it's take single (r,g,b) value"""
    try:
        if ( type_ == "fg" ):
            return "\033[{};38;2;{};{};{}m{}\033[0;38;2;255;255;255m".format(style,rgb[0], rgb[1], rgb[2], text)
        else:
            return "\033[{};48;2;{};{};{}m{}\033[0;48;0;255;255;255m".format(style,rgb[0], rgb[1], rgb[2], text)
            
    except:
        raise TypeError(" Unsupported input type it needs a tuple() ")
    
    
    
def image_art(Character : str, Path : str,Type = "fg") -> str :
    """ it's return the Coloerd text based on given image\n
    -------------------------------------------------------------------------------------------\n
        Note : it's Only get JPG File and using below ( 150 X 150 )px image is more suitable  |\n
    -------------------------------------------------------------------------------------------\n
        ex : image_art(Character : str, Path : str)\n
        Character = it takes string (or) character using this output are replicate given image\n
        Path = it's need Path of the file\n
        Type = it's need inputs like ForeGround["fg"] (or) BackGround["bg"] so based on that the output replicate\n
        for example : image_art( "*", /../../)
    """
    rgb : image_ =image_(Path)
    val : image_ = rgb.rgb_value()
    Size = rgb.img_size()
    try:
        print("prossing...")
        text_art : str = ""
        count : int = 0
        for i in range(Size[0]):
            text_art = text_art + "\n"
            for j in range(Size[1]):
                count : int = count + 1
                text_art : str = text_art + rgb_to_txt(Character,val[count-1],type_=Type)
        return text_art 
    except:
        raise TypeError(" Unsupported image type (or) file")

def rgb_to_name(rgb_val : tuple) -> Union[str,int] :
        """ there is the module under development so the rgb values in datas it's return name of the data else it return 0."""
        datas = tuple_name
        data=list(datas.keys())
        if rgb_val in data:
            return datas.get(rgb_val)
        else:
            return 0
        
def name_to_rgb(Name_ofthe_color : str ) -> tuple :
    """ there is the module under development so the given name in datas it's return value of the data else it return None."""
    return colors.get(Name_ofthe_color.upper())

def hex_to_name(Name_ofthe_color : str ) -> str:
    """ there is the module under development so the given hex in datas it's return value of the data else it return None."""
    return hex_name.get(Name_ofthe_color)
def name_to_hex(hex_val : str ) -> str:
    """ there is the module under development so the given name in datas it's return value of the data else it return None."""
    return name_hex.get(hex_val)

def rgb_to_hex(rgb_val : tuple ) -> str:
    """ there is the module under development so the given rgb in datas it's return value of the data else it return None."""
    return name_to_hex(rgb_to_name(rgb_val))

def hex_to_rgb(rgb_val : str ) -> tuple:
    """ there is the module under development so the given hex in datas it's return value of the data else it return None."""
    return name_to_rgb(hex_to_name(rgb_val))
    
